src/App.js

```diff
@@ -94,45 +131,80 @@ class App extends React.Component {
     this.startEdit = true;
   }
 
+  /**
+   * set edit object which being edited to null so that value can be changed in input field
+   * @returns {undefined}
+   */
   editAlreadyUsed = () => {
     this.setState({ edit: null });
   }
 
-  render() {
+  componentDidMount() {
+    //if data is already in local storage use it
+    const storageData = window.localStorage.getItem('todoData');
+    const todos = storageData ? JSON.parse(storageData) : [];
+    this.setState({
+      todos
+    })
 
-    window.localStorage.clear();
-    window.localStorage.setItem('todoData', JSON.stringify(this.state.todos));
+    //stores track of last given id for todo item
+    this.idGenerate = parseInt(window.localStorage.getItem('idGenerator')) || 0;
 
-    const todos = this.state.todos;
+    //store data when tab is closing
+    window.addEventListener('beforeunload', (e) => {
+      window.localStorage.clear();
+      window.localStorage.setItem('idGenerator', this.idGenerate.toString());
+      window.localStorage.setItem('todoData', JSON.stringify(this.state.todos));
+    })
+  }
 
-    let className = 'container-fluid wrapper';
+  render() {
+    const { todos } = this.state;
+    const editToogle = this.startEdit;
 
-    let editToogle = this.startEdit;
     this.startEdit = false;
+
     return (
-      <div className={className}>
-        <h1 className='app-name'>Todo App</h1>
+      <>
         <div className='myContainer'>
           <Tabs changeTab={this.changeTab} tab={this.state.tab} />
+
+          <InputBar
+            isSearch={true}
+            btnText={'Search'}
+            submit={this.search}
+            placeholderText={'Search here'}
+          />
+
           <InputBar
-            addTodo={this.addTodo}
+            submit={this.addTodo}
             edit={this.state.edit}
-            resetEdit={this.editAlreadyUsed}
             startEdit={editToogle}
-            btnText={this.state.editIndex != null ? 'Edit' : 'Add'}
+            resetEdit={this.editAlreadyUsed}
+            placeholderText={'Enter Todo Here'}
+            btnText={this.state.editIndex != null ? 'Save' : 'Add'}
           />
 
-          <TodoBox
+          <TodoList
             todos={todos}
-            changeCompletion={this.changeCompletion}
             filter={this.state.tab}
-            deleteTodoItem={this.deleteTodoItem}
+            search={this.state.search}
             editTodoItem={this.editTodoItem}
+            deleteTodoItem={this.deleteTodoItem}
+            changeCompletion={this.changeCompletion}
           />
         </div>
-
-      </div>
+      </>
     );
   }
 }
-export default App;
\ No newline at end of file
+
+const withAppTitle = (Component) =>
+  (props) => (
+    <div className={'container-fluid wrapper'}>
+      <h1 className='app-name'>Todo App</h1>
+      <Component {...props} />
+    </div>
+  )
+
+export default withAppTitle(App);
```

Comment:
.

Reply:
I was just referring to add extra line at end.

![image](https://user-images.githubusercontent.com/25525098/51070816-8af4f100-166f-11e9-9207-a00059853cf1.png)
Hover on the red icon.